### PR TITLE
remove use of other bundles parameters in dependency injection extension

### LINKED ISSUE #166: Bundle may not work dependeing on other bundles loading order
While using SonataCacheBundle in a Flex application, I get an error about missing `doctrine.connections`:
```
  [Symfony\Component\DependencyInjection\Exception\ParameterNotFoundException]
  You have requested a non-existent parameter "doctrine.connections".


Exception trace:
 () at .../vendor/symfony/dependency-injection/ParameterBag/ParameterBag.php:100
 Symfony\Component\DependencyInjection\ParameterBag\ParameterBag->get() at .../vendor/symfony/dependency-injection/ParameterBag/EnvPlaceholderParameterBag.php:56
 Symfony\Component\DependencyInjection\ParameterBag\EnvPlaceholderParameterBag->get() at .../vendor/symfony/dependency-injection/Container.php:146
 Symfony\Component\DependencyInjection\Container->getParameter() at .../vendor/sonata-project/cache-bundle/DependencyInjection/SonataCacheExtension.php:97
 Sonata\CacheBundle\DependencyInjection\SonataCacheExtension->configureORM() at .../vendor/sonata-project/cache-bundle/DependencyInjection/SonataCacheExtension.php:62
 Sonata\CacheBundle\DependencyInjection\SonataCacheExtension->load() at .../vendor/symfony/dependency-injection/Compiler/MergeExtensionConfigurationPass.php:66
 Symfony\Component\DependencyInjection\Compiler\MergeExtensionConfigurationPass->process() at .../vendor/symfony/http-kernel/DependencyInjection/MergeExtensionConfigurationPass.php:39
 Symfony\Component\HttpKernel\DependencyInjection\MergeExtensionConfigurationPass->process() at .../vendor/symfony/dependency-injection/Compiler/Compiler.php:143
 Symfony\Component\DependencyInjection\Compiler\Compiler->compile() at .../vendor/symfony/dependency-injection/ContainerBuilder.php:746
 Symfony\Component\DependencyInjection\ContainerBuilder->compile() at .../vendor/symfony/http-kernel/Kernel.php:575
 Symfony\Component\HttpKernel\Kernel->initializeContainer() at .../vendor/symfony/http-kernel/Kernel.php:117
 Symfony\Component\HttpKernel\Kernel->boot() at .../vendor/symfony/framework-bundle/Console/Application.php:65
 Symfony\Bundle\FrameworkBundle\Console\Application->doRun() at .../vendor/symfony/console/Application.php:125
 Symfony\Component\Console\Application->run() at .../bin/console:36
```

This occurs because the [SonataCacheExtension](https://github.com/sonata-project/SonataCacheBundle/blob/e1db36ce6fc9b37354706a3af77b6e811aee2f56/DependencyInjection/SonataCacheExtension.php#L93-L101) use a parameter registered by `DoctrineBundle`.

As we are adding the subscriber to every connection, I suggest to remove this code and add the `doctrine.event_subscriber` to the `sonata.cache.orm.event_subscriber` definition without the `connection` attribute.

Same change could be applied to the `sonata.cache.phpcr_odm.event_subscriber` definition.

If you agree with those changes, I will open a PR.

### PR DESCRIPTION
**Pull request by fbourigault**
I'm targeting `2.x` branch because it does not introduce any backward incompatible change.

Closes #166 

## Changelog
```markdown
### Fixed
- DoctrineBundle and DoctrinePHPCRBundle can be loaded before or after SonataCacheBundle.
```

## To do
- [x] Add tests

## Subject

This will improve DX when bundle is used with Symfony Flex. I wanted to add tests but this bundle does not depend on `symfony/framework-bundle`. Can I add it to dependencies and write some tests?

### CODE REVIEW

**Comment by greg0ire** [comment]
You're more than welcome to do that. You mean dev requirements, right?

**Comment by fbourigault** [comment]
I noticed there are missing dependencies because the `symfony/dependency-injection` was not installed. I will try first by adding `symfony/dependency-injection` as regular dependency. Maybe it's enough.

**Comment by greg0ire** [comment]
OK great !

File: DependencyInjection/SonataCacheExtension.php [hunk 1]
```diff
@@ -89,34 +93,6 @@ public function configureInvalidation(ContainerBuilder $container, $config)
     /**
      * @param ContainerBuilder $container
      * @param array            $config
-     */
-    public function configureORM(ContainerBuilder $container, $config)
```
Comment (core23) [inline DependencyInjection/SonataCacheExtension.php]: BC break

**Comment by fbourigault** [comment]
I just pushed a rebased BC version of this PR.

**Comment by fbourigault** [comment]
That's not so hard. I just need to add `symfony/dependency-injection` as dependency. Should I add it as `require` or `require-dev`. Choosing the second over the first means allowing this repository not to be used as a Symfony bundle. Tell me which option to choose and I will add some tests.

**Comment by greg0ire** [comment]
"bundle" is in the name of the package, so please make it a hard dependency.

**Comment by fbourigault** [comment]
I added tests to assert that listeners are loaded when doctrine/orm and doctrine/phpcr-odm packages are installed. This revealed a typo in the `Doctrine\ODM\PHPCR\Version` class name.

File: tests/DependencyInjection/SonataCacheExtensionTest.php [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\CacheBundle\Tests\DependencyInjection\Configuration;
+
+use PHPUnit\Framework\TestCase;
+use Sonata\CacheBundle\DependencyInjection\SonataCacheExtension;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
```
Comment (OskarStark) [inline tests/DependencyInjection/SonataCacheExtensionTest.php]: please add `@author` tag

**Comment by greg0ire** [comment]
Thanks @fbourigault !

**Pull request merged**